### PR TITLE
Improve Kedro-Viz docs for Experiment Tracking

### LINKED ISSUE #2144: Improve Kedro-Viz documentation for Experiment Tracking
This issue is to track the work on the third and final page of the Kedro-Viz section of docs which discusses Experiment Tracking.: https://kedro.readthedocs.io/en/stable/visualisation/experiment_tracking.html

There are a raft of changes/improvements coming and this ticket should cover any rework of the docs to include those.

I suggest that we revise the page to make it more of a tutorial in experiment tracking (maybe taking the Spaceflights starter and extending, but it could be a separate example). We could potentially create a separate page for information about using the experiment tracking feature, and keep the tutorial about it separate. Let's see how much copy there is as the page is currently too long.

### PR DESCRIPTION
**Pull request by tynandebold**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.

## Description

Resolves (part of) #2144.

> **Warning**
> This PR shouldn't be merged until version 5.2.0 of Kedro-Viz is released. This is because there's content in this PR that will only exist when that release is out.

## Development notes

Taken from the comment [here](https://github.com/kedro-org/kedro/issues/2144#issuecomment-1375703174), I've done the following:

- [x] Ensure that the spaceflights tutorial works without issue. There was a problem with Plotly where the charts weren't rendering (relates to https://github.com/kedro-org/kedro-viz/issues/1205).
- [x] There are actually two pages for exp. tracking documentation. We should only have one.
    - Copy the contents of the second, smaller page into the first and then delete that content from the second page.
- [x] Mention parallel coordinate and time-series plots.

A note on each checked item:

1. I checked this using Kedro 0.18.3, since 0.18.4 will prevent the charts from loading. There were a couple of minor hangups while running through the tutorial that I've remedied. All should work smoothly now.
2. I've deleted this [secondary docs page](https://kedro.readthedocs.io/en/latest/logging/experiment_tracking.html) with experiment tracking instructions and didn't move anything over into the main page because it was already well covered.
3. The new work is captured and takes precedent in the ordering on the page, meaning the metrics plots in the experiment tracking page shows up earlier in the documentation versus the metrics plot that can be viewed on the flowchart page.

Lastly, I've updated the gifs and images to reflect the changes in design on the experiment tracking page (now we have tabs for overview, metrics, and plots, and before we didn't).

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/2193"></a>

### CODE REVIEW

File: docs/source/visualisation/experiment_tracking.md
Comment (tynandebold) [inline docs/source/visualisation/experiment_tracking.md:1]: I've updated the wording here to be consistent with the other titles of the sections under `Visualisation with Kedro-Viz`.

File: docs/source/visualisation/experiment_tracking.md
Comment (tynandebold) [inline docs/source/visualisation/experiment_tracking.md:200]: This import was missing. Without it, doing a `kedro run` failed.

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:9]: I'd be tempted to make this more imperative.

```suggestion
To enable the experiment tracking features of Kedro-Viz you need to:
```

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:11]:
```suggestion
- [Set up a session store to capture experiment metadata](#set-up-the-session-store),
```

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:12]:
```suggestion
- [Set up experiment tracking datasets to list the metrics to track](#set-up-tracking-datasets),
```

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:13]:
```suggestion
- [Modify your nodes and pipelines to output those metrics](#set-up-your-nodes-and-pipelines-to-log-metrics).
```

**Comment by stichbury** [comment]
This looks great, thanks @tynandebold.

I do have a query about Spaceflights. So this now works if you take the most recent spaceflights starter and add the code given above? There's no reliance on any previous part of the Kedro-Viz docs like the previous page on Plotly reporting? I would normally test it myself just to 'dogfood' the instructions, but I'm a bit tied up with the Academy decks today and don't want to hold you up.

I think we still need to introduce Kedro-Viz for anyone (unlikely enough) to be coming straight to this page without having previously installed Viz. [We put this into the Plotly page.](https://github.com/kedro-org/kedro/blob/main/docs/source/visualisation/visualise_charts_with_plotly.md#visualisation-with-plotly) -- would it be possible to add to this file?

**Comment by stichbury** [comment]
This looks great to me @tynandebold. LMK when you've made those changes and I'll approve.

**Comment by tynandebold** [comment]
I've made the changes now @stichbury. I took your second suggestion and moved the testing of the Viz version to after the install step. Let me know how that reads and if it's in the correct place.

And to your query about Spaceflights: yes, all worked fine for me when I tested with the latest version of Spaceflights (using the `kedro new` command) and added the code from this PR. I used a fresh conda environment for testing this and followed the instruction from this page only, taking nothing from any other page in the docs. 

Could there have been some leftover dependencies hanging around in my new conda environment that were populated from an older one? I'm not too familiar with this. Again, it worked fine for me with 0.18.3 of Kedro and the couple new lines that I added, so I'm quite confident it's solid. I do hope to test with Merel's fix though, to ensure it'll work with 0.18.4.

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:13]:
```suggestion
- [set up a session store to capture experiment metadata](#set-up-the-session-store),
- [set up experiment tracking datasets to list the metrics to track](#set-up-tracking-datasets),
- [modify your nodes and pipelines to output those metrics](#set-up-your-nodes-and-pipelines-to-log-metrics).
```

Comment (stichbury) [reply docs/source/visualisation/experiment_tracking.md:13]: Sorry, @tynandebold This was my introduction -- I forgot the guidelines for bullets 🤦

**Review by stichbury — APPROVED** [review]
LGTM, thanks for this! 🌟 🌟 🌟 🌟 🌟

File: docs/source/visualisation/experiment_tracking.md
Comment (tynandebold) [reply docs/source/visualisation/experiment_tracking.md]: Now I'm confused. Should they be capitalized or lowercase?

Comment (stichbury) [reply docs/source/visualisation/experiment_tracking.md]: You know what, I'll roll it back to capitals, since other pages of the docs do have them capitalised. I've no idea when/if that's a change, as I may be thinking of another style guide, but that doesn't explain why the bullets in the original copy were lower case here. Never mind. I'm now happy for you to merge this when you are ready.

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:11]:
```suggestion
- [Set up a session store to capture experiment metadata](#set-up-the-session-store),
```

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:12]:
```suggestion
- [Set up experiment tracking datasets to list the metrics to track](#set-up-tracking-datasets),
```

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:13]:
```suggestion
- [Modify your nodes and pipelines to output those metrics](#set-up-your-nodes-and-pipelines-to-log-metrics).
```

File: docs/source/visualisation/experiment_tracking.md
Comment (tynandebold) [reply docs/source/visualisation/experiment_tracking.md:13]: Copy that!

**Review by merelcht — APPROVED** [review]
This looks good to me! And so exciting to see the time series and parallel coordinates! 🥳

File: docs/source/visualisation/experiment_tracking.md
Comment (yetudada) [inline docs/source/visualisation/experiment_tracking.md:11]: I will need help from @stichbury and @NeroOkwa in this new section. Does this content work?

File: docs/source/visualisation/experiment_tracking.md
Comment (yetudada) [inline docs/source/visualisation/experiment_tracking.md:73]: My only other comment is based on this and it might need to be a bigger change to how we structure this tutorial. We introduce namespacing here and in the plot comparison section (`data_processing.confusion_matrix` in `catalog.yml`) but we don't actually explain what the full stop means here. 

It's important that we do because I got confused with the guidance in 113 - `Note that the output dataset must exactly match the name of the tracking dataset specified in the catalog file.` because users are asked to just specify `metrics` and not `active_modelling_pipeline.metrics`. 

So my question would be: 
 - Can we, ideally, do this tutorial without introducing the concept of namespaces too (it's an information overload)?
 - And if not, how do we introduce it?

File: docs/source/visualisation/experiment_tracking.md
Comment (yetudada) [inline docs/source/visualisation/experiment_tracking.md:126]:
```suggestion
Repeat the same steps to set up the `companies_column` dataset. For this dataset, log the column that contains the list of companies as outlined in the `companies.csv` file under the `data/01_raw` directory. Modify the `preprocess_companies` node under the `data_processing` pipeline (`src/kedro-experiment-tracking-tutorial/pipelines/data_processing/nodes.py`) to return the data under a key-value pair, as shown below:
```

**Review by yetudada — CHANGES_REQUESTED** [review]
Thank you so much for this stellar work @tynandebold! I've made some additions and I've left a bigger question which we might need @rashidakanchwala or @merelcht to help with.

File: docs/source/visualisation/experiment_tracking.md
Comment (merelcht) [reply docs/source/visualisation/experiment_tracking.md:73]: Great spot @yetudada ! We do indeed not need the namespace here, as no namespacing has been applied to the pipelines.

Comment (yetudada) [reply docs/source/visualisation/experiment_tracking.md:73]: Awesome let me update the other mentions of `active_modelling_pipeline.metrics` 😊

Comment (yetudada) [inline docs/source/visualisation/experiment_tracking.md:73]:
```suggestion
metrics:
```

File: docs/source/visualisation/experiment_tracking.md
Comment (yetudada) [inline docs/source/visualisation/experiment_tracking.md:70]:
```suggestion
Set up two datasets to log the columns used in the companies dataset (`companies_columns`) and experiment metrics for the data science pipeline (`metrics`) like the coefficient of determination (`r2 score`),  max error (`me`) and mean absolute error (`mae`) by adding the following in the `conf/base/catalog.yml` file:
```

File: docs/source/visualisation/experiment_tracking.md
Comment (yetudada) [reply docs/source/visualisation/experiment_tracking.md:11]: From @NeroOkwa: "yeah looks great!"

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [inline docs/source/visualisation/experiment_tracking.md:16]: I have a bit of a problem with "Experiment tracking in Kedro is a visual upgrade of your experience using MLFlow Tracking. Leverage `kedro-mlflow` for your workflow. " as it sounds like marketing and doesn't explain how Kedro upgrades MLFlow Tracking visually.

Can I suggest:

```suggestion
 - **MLflow** - You can combine MLFlow with Kedro by using [`kedro-mlflow`](https://kedro-mlflow.readthedocs.io/en/stable/) if you require experiment tracking, model registry and/or model serving capabilities or have access to Managed MLflow within the Databricks ecosystem.  
```

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [reply docs/source/visualisation/experiment_tracking.md:11]: Yep, looks good to me. I've made one suggestion but it's not a huge issue so you can leave as you've edited if it's simpler.

**Comment by stichbury** [comment]
> I think we just need a general check to see if this tutorial runs from beginning to end in a clean virtual environment with all the changes but otherwise I think we're ready to go live.

I'm happy to try this out as I want to test it out and learn about experiment tracking, but I'm not really able to right now. I think @tynandebold needs to hold off until the next viz release anyway, so hopefully this can wait until next week -- I'll get it done before the end of the sprint.

**Comment by tynandebold** [comment]
@stichbury you'll have a couple days to test it out. Hopefully we release Viz tomorrow or Wednesday.

Separately, in this PR I think we should remove the references to Kedro version `0.18.3` [in this file](https://github.com/kedro-org/kedro/blob/main/docs/source/visualisation/visualise_charts_with_plotly.md), since that problem should be solved when this [PR is completed](https://github.com/kedro-org/kedro-viz/pull/1214). What do you think?

File: docs/source/visualisation/experiment_tracking.md
Comment (stichbury) [reply docs/source/visualisation/experiment_tracking.md:200]: True! I have to make some wholesale changes to the other viz docs when that PR merges so can remove this then if you merge this PR ahead of #1214 but otherwise, if it won't merge in until later, you can certainly remove this.

**Comment by stichbury** [comment]
> @stichbury you'll have a couple days to test it out. Hopefully we release Viz tomorrow or Wednesday.
> 
> Separately, in this PR I think we should remove the references to Kedro version `0.18.3` [in this file](https://github.com/kedro-org/kedro/blob/main/docs/source/visualisation/visualise_charts_with_plotly.md), since that problem should be solved when this [PR is completed](https://github.com/kedro-org/kedro-viz/pull/1214). What do you think?

I'll prioritise this on Tuesday 17th.

File: docs/source/visualisation/experiment_tracking.md
Comment (tynandebold) [reply docs/source/visualisation/experiment_tracking.md]: I've removed it since this PR will be merged in the next couple of days.

**Comment by stichbury** [comment]
> > @stichbury you'll have a couple days to test it out. Hopefully we release Viz tomorrow or Wednesday.
> > Separately, in this PR I think we should remove the references to Kedro version `0.18.3` [in this file](https://github.com/kedro-org/kedro/blob/main/docs/source/visualisation/visualise_charts_with_plotly.md), since that problem should be solved when this [PR is completed](https://github.com/kedro-org/kedro-viz/pull/1214). What do you think?
> 
> I'll prioritise this on Tuesday 17th.

@tynandebold I've now tested the text and all is good. I made a couple of changes to the experiment tracking page to tweak it (nothing major) and also removed mention of Kedro 0.18.3 in the Viz docs about plotly charts, which I've sneaked into this PR. Hope that's OK!

**Comment by tynandebold** [comment]
Fantastic, @stichbury! Thank you 💥

**Pull request merged**
